pydantic_settings/main.py

```diff
@@ -90,38 +94,46 @@ def settings_customise_sources(
     def _settings_build_values(
         self,
         init_kwargs: dict[str, Any],
+        _case_sensitive: bool | None = None,
+        _env_prefix: str | None = None,
         _env_file: DotenvType | None = None,
         _env_file_encoding: str | None = None,
         _env_nested_delimiter: str | None = None,
         _secrets_dir: str | Path | None = None,
     ) -> dict[str, Any]:
+        # Determine settings config values
+        case_sensitive = _case_sensitive if _case_sensitive is not None else self.model_config.get('case_sensitive')
+        env_prefix = _env_prefix if _env_prefix is not None else self.model_config.get('env_prefix')
+        env_file = _env_file if _env_file != env_file_sentinel else self.model_config.get('env_file')
+        env_file_encoding = (
+            _env_file_encoding if _env_file_encoding is not None else self.model_config.get('env_file_encoding')
+        )
+        env_nested_delimiter = (
+            _env_nested_delimiter
+            if _env_nested_delimiter is not None
+            else self.model_config.get('env_nested_delimiter')
+        )
+        secrets_dir = _secrets_dir or self.model_config.get('secrets_dir')
```

Comment:
Should we use the `_val or self.model_config.get(val)` form for each of this settings instead, as it is done for `secrets_dir`? (kept it the way it was for each value)

Reply:
Yes

Reply:
In fact better to keep the strict `None` equality check, otherwise this could lead to unexpected results (e.g. with `_case_sensitive=False`, `case_sensitive = _case_sensitive or self.model_config.get('case_sensitive')` would evaluate to `self.model_config.get('case_sensitive')`).